File: src/main/java/uss/code/member/dto/response/MemberProfileResponse.java
package uss.code.member.dto.response;

import lombok.Builder;
import uss.code.member.domain.Member;

import static lombok.AccessLevel.PRIVATE;

@Builder(access = PRIVATE)
public record MemberProfileResponse(
        String email,

        String department,

        String studentId,

        String name,

        String grade,

        String academicStatus,

        double gpa,

        int creditLimit
) {
    public static MemberProfileResponse of(final Member member) {
        return MemberProfileResponse.builder()
                .email(member.getEmail())
                .department(member.getDepartment().getName())
                .studentId(member.getStudentId())
                .name(member.getName())
                .grade(member.getGrade().getName())
                .academicStatus(member.getAcademicStatus().getName())
                .gpa(member.getLastSemesterGpa())
                .creditLimit(member.getMaxCredit())
                .build();
    }
}

File: src/main/kotlin/uss/code/member/dto/response/MemberProfileResponse.kt
package uss.code.member.dto.response

import uss.code.member.domain.Member

data class MemberProfileResponse(
    val email: String,

    val department: String,

    val studentId: String,

    val name: String,

    val grade: String,

    val academicStatus: String,

    val gpa: Double,

    val creditLimit: Int,
) {
    companion object {
        fun of(member: Member): MemberProfileResponse {
            return MemberProfileResponse(
                email = member.email,
                department = member.department.displayName,
                studentId = member.studentId,
                name = member.name,
                grade = member.grade.displayName,
                academicStatus = member.academicStatus.displayName,
                gpa = member.lastSemesterGpa,
                creditLimit = member.maxCredit,
            )
        }
    }
}
